### PR TITLE
reference/sql: update data types

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

- add ref in `json.md`
- align `data and time.md` with docs
- update TOC.md
- remove redundant `>` in `numeric.md`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/2257
- Other reference link(s):

### CODE REVIEW

File: reference/sql/data-types/date-and-time.md [hunk 1]
```diff
@@ -33,11 +303,11 @@ TIME[(fsp)]
 
 > **注意：**
 >
-> 注意 `TIME` 的缩写形式。例如，'11:12' 表示 '11:12:00' 而不是 '00:11:12'。但是，'1112' 表示 '00:11:12'。这些差异取决于 `:` 字符的存在与否。
+> 注意 `TIME` 的缩写形式。例如，`11:12` 表示 `11:12:00` 而不是 `00:11:12`。但是，`1112` 表示 `00:11:12`。这些差异取决于 `:` 字符的存在与否。
 
 ### `DATETIME` 类型
 
-`DATETIME` 类型是日期和时间的组合，格式为 `YYYY-MM-DD HH:MM:SS[.fraction]`。支持的范围是 `1000-01-01 00:00:00.000000` 到 `9999-12-31 23:59:59.000000`。`fsp` 参数表示秒精度，取值范围为 0~6，默认值为 0。
+`DATETIME` 类型是日期和时间的组合，格式为 `YYYY-MM-DD HH:MM:SS[.fraction]`。支持的范围是 `1000-01-01 00:00:00.000000` 到 `9999-12-31 23:59:59.999999`。`fsp` 参数表示秒精度，取值范围为 0~6，默认值为 0。TiDB 支持字符串或数字转换为 `DATETIME` 类型。
```
Comment (TomShawn) [inline reference/sql/data-types/date-and-time.md:310]: @toutdesuite This section is a little bit different from the English version
image PTAL

File: reference/sql/data-types/date-and-time.md [hunk 2]
```diff
@@ -47,20 +317,38 @@ DATETIME[(fsp)]
...
 {{< copyable "sql" >}}
 
 ```sql
 YEAR[(4)]
 ```
+
+`YEAR` 类型遵循以下格式规则:
+
++ 如果是四位数的数值，支持的范围是 `1901` 至 `2155`。
++ 如果是四位数的字符串，支持的范围是 `'1901'` 到 `'2155'`。
```
Comment (TomShawn) [inline reference/sql/data-types/date-and-time.md:349]:
```suggestion
+ 如果是四位数的字符串，支持的范围是 `1901` 到 `2155`。
```

File: reference/sql/data-types/date-and-time.md (hunk 1 above)
Comment (toutdesuite) [reply reference/sql/data-types/date-and-time.md:310]: @qw4990 Please confirm whether the valid values range is to `9999-12-31 23:59:59.000000` or to `9999-12-31 23:59:59.999999`. Thanks!

Comment (qw4990) [reply reference/sql/data-types/date-and-time.md:310]: After testing, the actual maximum value of both `MySQL` and `TiDB` is `9999-12-31 23:59:59.499999`, but I think this is a bug.
According to [MySQL’s document](https://dev.mysql.com/doc/refman/8.0/en/datetime.html), the maximum value should be `9999-12-31 23:59:59.999999`, so I prefer to use `9999-12-31 23:59:59.999999` here.

**Pull request merged**